### PR TITLE
Set param_scan_axis=0 and change all checkpoint creation files

### PR DESCRIPTION
**Pull request by khatwanimohit**
# Description

Setting param_scan_axis=0 helps in improving perf and reducing memory required by optimizer state.

FIXES: #1382 

# Tests

Since this is touching all the checkpoint creation files, I'm going to ask help for creating new checkpoints and comparing logits.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ ] I have performed a self-review of my code.
- [ ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ ] I have run end-to-end tests tests and provided workload links above if applicable.
- [ ] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/configs/inference.yml [hunk 1]
```diff
@@ -203,7 +203,7 @@ out_proj: 'remat'
 
 optimizer_memory_host_offload: False
 scan_layers: True # We recommend setting this to false when using pipeline parallelism, instead scanning the PP iterations.
-param_scan_axis: 1
+param_scan_axis: 0
```
Comment (khatwanimohit) [inline MaxText/configs/inference.yml:206]: @vipannalla is this change okay? see the issue linked in the description

**Review by khatwanimohit — COMMENTED** [review]
I want to give a shout out to @ZhiyuLi-goog :) who wrote the GPT3 checkpoint converter  really nice and configurable so that it works even with changes like this. Impressive!

eg: `"['decoder']['layers']['pre_self_attention_norm']['bias']": (
          ".params.lm.transformer.repeat.sub.x_layers_0.layer_norm.bias",
          lambda x: np.moveaxis(x, 0, cfg.param_scan_axis),
      ),`
